git_pile/git_pile.py

```diff
@@ -1189,22 +1189,23 @@ def cmd_genbranch(args):
     # work in a separate directory to avoid cluttering whatever the user is doing
     # on the main one
     with temporary_worktree(baseline, root) as d:
+        branch = args.branch if args.branch else config.result_branch
+        path = git_worktree_get_checkout_path(root, branch)
+
+        if path and not args.force:
+            error("can't use branch '%s' because it is checked out at '%s'" % (branch, path))
+            return 1
+
         git(["-C", d] + apply_cmd + patchlist, stdout=stdout)
 
         if args.dirty:
             raise temporary_worktree.Break
 
         head = git(["-C", d, "rev-parse", "HEAD"]).stdout.strip()
 
-        branch = args.branch if args.branch else config.result_branch
-        path = git_worktree_get_checkout_path(root, branch)
         if path:
-            if not args.force:
-                error("final result is %s but branch '%s' could not be updated to it "
-                      "because it is checked out at '%s'" % (head, branch, path))
```

Comment:
btw the intention of this error message here was to cover this situation "oh, I waited that long for the genbranch, but the branch could not be updated". The commit hash in "final result is <hash>" is just the commit you could then use as the final state, either to checkout a new branch or to reset your previous branch to it.